PCbuild/pyproject.props

```diff
@@ -12,6 +12,7 @@
     <IntDir>$(IntDir.Replace(`\\`, `\`))</IntDir>
     <!-- pyconfig.h is updated by pythoncore.vcxproj, so it's always in pythoncore's IntDir -->
     <GeneratedPyConfigDir>$(Py_IntDir)\$(MajorVersionNumber)$(MinorVersionNumber)$(ArchName)_$(Configuration)\pythoncore\</GeneratedPyConfigDir>
+    <GeneratedFrozenModulesDir>$(Py_IntDir)\frozen\</GeneratedFrozenModulesDir>
```

Comment:
Let's put this inside the `MajorVersionNumber...` segment from the one above. That way it's easier to switch between branches without having to recompile everything (in theory the architecture and configuration are irrelevant, but I suspect it will cause more rebuilds if we don't separate the intermediate files based on them).

A worthwhile test is to clean everything (`rmdir PCbuild/obj`) and then build twice. The second build should skip all the compile and link steps (try passing `-v` to see). Switching between debug and release build (`build.bat [-d]`) and platform (`build.bat -p x64/x86`) shouldn't force everything to rebuild.

If you switch branch, build, and then switch back, _ideally_ we'd still skip all the steps, but it'll depend on whether Git preserves modification dates property. But it certainly shouldn't build with the wrong files, which is the bigger risk.

Reply:
I updated the PR to use a version+arch+config segment as you suggested, and I think something weird is going on...

the x86 build succeeded, but the x64 build fails.
the x64 failure is due to [failing to find "getpath.h"](https://github.com/python/cpython/actions/runs/7309700119/job/19917585388?pr=113303#step:4:572)
the [compile command](https://github.com/python/cpython/actions/runs/7309700119/job/19917585388?pr=113303#step:4:570) looks correct (in particular, it contains a correct-looking include dir flag, `/ID:\a\cpython\cpython\PCbuild\obj\\313amd64_Debug\frozen\`)), and [getpath.h is generated earlier in the build](https://github.com/python/cpython/actions/runs/7309700119/job/19917585388?pr=113303#step:4:398).
but taking a closer look at the getpath.h generation steps, copied here verbatim for convenience:

```
       _RebuildGetPath:
         "D:\a\cpython\cpython\PCbuild\win32\_freeze_module_d.exe" "getpath" "D:\a\cpython\cpython\Modules\getpath.py" "D:\a\cpython\cpython\PCbuild\obj\313win32_Debug\_freeze_module\getpath.g.h"
         Copying file from "D:\a\cpython\cpython\PCbuild\obj\313win32_Debug\_freeze_module\getpath.g.h" to "D:\a\cpython\cpython\PCbuild\obj\313win32_Debug\frozen\Python\frozen_modules\getpath.h".
```

notice how this looks like the win32 configuration, not the amd64 one! the `_freeze_module` binary is coming from `PCbuild\313win32`, and the getpath input and output files are in the `PCbuild\obj\313win32_Debug` directory, where I'd expect "amd64" instead of "win32".

I believe this is a pre-existing issue, based on the observation that [a recent x64 build on the main branch](https://github.com/python/cpython/actions/runs/7309482675/job/19917143777#step:4:397) shows the same pattern (of using the "win32" build dirs), but it doesn't cause any issues because the generated file ends up in the source tree.

I am not sure why this is happening this way, or how to fix it.

Reply:
found the issue.
in `pcbuild.proj`, the `FreezeProjects` group has this:

```
<Platform>$(PreferredToolArchitecture)</Platform>
```

while the other project groups has this:

```
<Platform>$(Platform)</Platform>
```

so if I use `$(PreferredToolArchitecture)` in `GeneratedFrozenModulesDir` it works (and shares the frozen dir between x86 and x64, which I think is fine, and is consistent with how things worked before this change, e.g. in the source tree).

Reply:
Maybe instead we should just replace the `$(Platform)` with `frozen` and not have another subdirectory? It's only the generated files, so switching `PreferredToolArchitecture` shouldn't matter, but it's possible that one day we'll need to set a preference for x64 tools for one project, and totally omitting the platform here will save us diagnosing a new problem in the future.

Reply:
something like [this](https://github.com/python/cpython/pull/113303/commits/851fdce458edb879a2653013dcd7052988f3cc77)?